### PR TITLE
feat(outfitter): DX migration — action-registry scanner, codemods, --example, enhanced test helpers

### PR DESCRIPTION
**Pull request by galligan**
## Summary

- Add `check action-registry` scanner for command/action completeness
- Enhance `testCommand()` / `testTool()` with schema, context, and hints support
- Add `upgrade codemod` for Commander-to-builder pattern transformation
- Add `--example` flag to `init` for pattern-rich scaffolding
- Add v0.5 migration guide with before/after for all breaking changes
- Fix registry scanner and testCommand env restoration
- Fix 4 codemod issues (arguments, optional semantics, edge cases)
- Fix 2 factual inaccuracies in migration-v0.5.md
- Update README for v0.5 Builder Pattern changes

## Linear Issues

Closes OS-235, Closes OS-351, Closes OS-352, Closes OS-353

## Test Plan

- [x] Action registry scanner tests
- [x] Codemod transformation tests
- [x] Enhanced test helper tests
- [x] `bun test` passes across affected packages

### CODE REVIEW

File: plugins/outfitter/shared/codemods/cli/commander-to-builder.ts [hunk 1]
```diff
@@ -0,0 +1,671 @@
...
+
+// =============================================================================
+// File Transformation
+// =============================================================================
+
+interface TransformResult {
+  readonly changed: boolean;
+  readonly content: string;
+  readonly hasPositionalArgs: boolean;
+}
```
[the remark this answers is not in the document]
Comment (galligan) [reply plugins/outfitter/shared/codemods/cli/commander-to-builder.ts]: Removed dead hasPositionalArgs from TransformResult and corresponding return paths.
